doc/src/tutorial/basic_operations.rst

```diff
@@ -11,6 +11,51 @@ in the :ref:`advanced expression manipulation <tutorial-manipulation>` section.
     >>> from sympy import *
     >>> x, y, z = symbols("x y z")
 
+Expressions can be compared using a regular python syntax::
+
+    >>> from sympy.abc import x, y
+    >>> x + y == y + x
+    True
+
+    >>> x + y == y - x
+    False
+
+We made the following decision in SymPy: ``a = Symbol("x")`` and another
+``b = Symbol("x")`` (with the same string "x") is the same thing, i.e ``a == b`` is
+``True``. We chose ``a == b``, because it is more natural - ``exp(x) == exp(x)`` is
+also ``True`` for the same instance of ``x`` but different instances of ``exp``,
+so we chose to have ``exp(x) == exp(x)`` even for different instances of ``x``.
+
+Sometimes, you need to have a unique symbol, for example as a temporary one in
+some calculation, which is going to be substituted for something else at the
+end anyway. This is achieved using ``Dummy("x")``. So, to sum it
+up::
+
+    >>> from sympy import Symbol, Dummy
+    >>> Symbol("x") == Symbol("x")
+    True
+
+    >>> Dummy("x") == Dummy("x")
+    False
+
+For computation, all expressions need to be in a canonical form, this is done
+during the creation of the particular instance and only inexpensive operations
+are performed, necessary to put the expression in the canonical form.
+Whenever you construct an expression, for example ``Add(x, x)``, the
+``Add.__new__()`` is called and it determines what to return. In this case::
+
+    >>> from sympy import Add
+    >>> from sympy.abc import x
+    >>> e = Add(x, x)
+    >>> e
+    2*x
+
+    >>> type(e)
+    <class 'sympy.core.mul.Mul'>
+
+``e`` is actually an instance of ``Mul(2, x)``, because ``Add.__new__()``
+returned ``Mul``.
+
```

Comment:
I would just remove this whole thing. It's confusing, and I think it's explained better in the advanced expression manipulation section of the tutorial (but let me know if you agree).